### PR TITLE
feat(storage): create a signed upload policy for uploading objects

### PR DESCRIPTION
**Pull request by athakor**
Implement as per python [Bucket.generate_upload_policy](https://github.com/googleapis/google-cloud-python/blob/ee6214251fbe83cb83a9b5b5019c605a27a01354/storage/google/cloud/storage/bucket.py#L1997-L2057), and a [snippet](https://github.com/googleapis/google-cloud-python/blob/ee6214251fbe83cb83a9b5b5019c605a27a01354/storage/docs/snippets.py#L253-L283) showing how it is used.

### CODE REVIEW

File: google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java [hunk 1]
```diff
@@ -40,15 +40,9 @@
...
 import java.security.Key;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.LinkedHashSet;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Set;
+import java.util.*;
```
Comment (pmakani) [inline google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java]: use single class setting in intellij

Comment (athakor) [reply google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java]: done

File: google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java [hunk 2]
```diff
@@ -129,6 +133,45 @@ public Bucket create(BucketInfo bucketInfo, BucketTargetOption... options) {
...
+    ServiceAccountSigner credentials =
+        (ServiceAccountSigner) optionMap.get(PolicyDocumentOption.Option.SERVICE_ACCOUNT_CRED);
+    if (credentials == null) {
+      checkState(
+          this.getOptions().getCredentials() instanceof ServiceAccountSigner,
+          "Signing key was not provided and could not be derived");
+      credentials = (ServiceAccountSigner) this.getOptions().getCredentials();
+    }
+    Map<String, String> bucketMap = ImmutableMap.of("bucket", bucket);
+    conditions.add(bucketMap);
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java]: `conditions.add(ImmutableMap.of("bucket", bucket));`

File: google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java [hunk 3]
```diff
@@ -1608,6 +1644,22 @@ public static Builder newBuilder() {
...
+  /**
+   * Create a signed upload policy for uploading objects.This method generates and signs a policy
+   * document. You can use `policy documents`_ to allow visitors to a website to upload files to
+   * Google Cloud Storage without giving them direct write access.
+   *
+   * @param bucket
+   * @param conditions
+   * @param expiration
+   * @param options
+   * @return
```
Comment (rahulKQL) [inline google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java]: either write some description here or remove the `@return` tag

Comment (athakor) [reply google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java]: done

File: google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java (hunk 2 above)
Comment (athakor) [reply google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java]: done

**Review by rahulKQL — APPROVED** [review]

**Pull request closed without merging**